### PR TITLE
update to `vite@2.7.x`

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Update our Vite branch to latest v2.7
- Useful for testing in this PR, and also may be worth merging for our own `next` release given the number of SSR fixes this version is reported to have

## Testing

 - Still needs more before merging

### CODE REVIEW

**Comment by natemoo-re** [comment]
Posted the first CI run to the Vite team to discuss.
https://discord.com/channels/804011606160703521/910615959922946148/910615961701326848

**Comment by drwpow** [comment]
I think you’ll need to update `astro/package.json`’s `vite` dependency to match the built version as well

File: packages/astro/src/vite-plugin-fetch/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { Plugin } from '../core/vite';
 import MagicString from 'magic-string';
 
 // https://github.com/vitejs/vite/discussions/5109#discussioncomment-1450726
-function isSSR(options: undefined | boolean | { ssr: boolean }): boolean {
+function isSSR(options: undefined | boolean | { ssr?: boolean }): boolean {
```
Comment (drwpow) [inline packages/astro/src/vite-plugin-fetch/index.ts:5]: This should probably go away entirely in `2.7`. The updated types have a different signature than when we added this.

**Comment by natemoo-re** [comment]
https://github.com/sveltejs/vite-plugin-svelte/pull/225 should unblock most of the issues here!

**Comment by boehs** [comment]
I would appreciate it if you waited until the next release, it allows you to do `?raw` in globEager

**Comment by natemoo-re** [comment]
This PR has gotten extremely difficult to rebase. Closing in favor of https://github.com/withastro/astro/pull/2424.

**Pull request closed without merging**